### PR TITLE
[boot0]:remove no need files before extract image

### PR DESCRIPTION
**Pull request by sihuihan88**
Signed-off-by: Sihui Han <sihan@microsoft.com>

### CODE REVIEW

**Comment by pavel-shirshov** [comment]
Why old code doesn't work anymore for us? Why do we need a new one?

Also `` (backticks) outdated, it's better to use $() for this.

**Comment by sihuihan88** [comment]
When we convert the arista device to sonic, there is a disk space issue since the old code can only remove the rw and docker folders but not the files related to arista. After discussed with Guohan, he suggests to remove all the no need files before extracting the image.

File: files/Aboot/boot0.j2 [hunk 1]
```diff
@@ -41,11 +41,13 @@ parse_environment_config() {
...
 
 extract_image() {
-    ## Clean old directory for read-write layer
-    rm -rf "$target_path/rw"
-
-    ## Clean docker directory
-    rm -rf "$target_path/{{ DOCKERFS_DIR }}"
+    ## Remove all the other unnecssary files except swi file, boot-config and boot0
+    for f in `ls -A $target_path`; do
+        if [ $f != "${swipath##*/}" ] && [ $f != "boot-config" ] && [ $f != "boot0" ]; then
```
Comment (lguohan) [inline files/Aboot/boot0.j2]: do we need to preserve boot0? boot0 is embedded in the swi file.

Comment (sihuihan88) [inline files/Aboot/boot0.j2]: Ah. If so, we don't need to preserve it here. Was thinking it locates in /mnt/flash lol.

**Comment by stcheng** [comment]
retest this please

**Comment by sihuihan88** [comment]
Retest this please

**Pull request merged**